packages/currency/src/currencyManager.ts

```diff
@@ -228,28 +228,53 @@ export class CurrencyManager<TMeta = unknown> implements ICurrencyManager<TMeta>
    * Throws if the currency is an ISO4217 currency.
    */
   static validateAddress(address: string, currency: CurrencyInput): boolean {
-    switch (currency.type) {
-      case RequestLogicTypes.CURRENCY.ISO4217:
-        throw new Error(`Could not validate an address for an ISO4217 currency`);
+    if (currency.type === RequestLogicTypes.CURRENCY.ISO4217) {
+      throw new Error(`Could not validate an address for an ISO4217 currency`);
+    }
+    return this.validateAddressWithNetworkAndType(address, currency.type, currency.network);
+  }
+
+  /**
+   * Validate the correctness of a Storage Currency
+   */
+  static validateCurrency(currency: StorageCurrency): boolean {
+    if (
+      currency.type === RequestLogicTypes.CURRENCY.ISO4217 ||
+      currency.type === RequestLogicTypes.CURRENCY.ETH ||
+      currency.type === RequestLogicTypes.CURRENCY.BTC
+    )
+      return true;
+    return this.validateAddressWithNetworkAndType(
+      currency.value,
+      currency.type,
+      currency.network as string,
+    );
+  }
+
+  /**
+   * Validates the correctness of an address based on a currency type and a network
+   */
+  static validateAddressWithNetworkAndType(
```

Comment:
maybe this should be an overload of `validateAddress` (which could be renamed 
`isValidAddress`)

Reply:
I removed this function and made the input of `validateAddress` an union (`ICurrencyInput | StorageCurrency`) instead.